### DISCUSSION TITLE
Compatible AF version for PIconnect 0.10.4

### DISCUSSION THREAD

**Discussion by Miigaa1191** [Hugovdberg/PIconnect#682] [Q&A]
First of all, thanks a lot for the 0.10.4 version upgrade. I'm able to pull `summaries` data but it doesn't work for my colleague's computer. It threw a ```TypeError error: int() argument must be a string, a bytes-loke object or a number, not 'AFSummaryTypes' ```. The code we run is: 

```
import PIconnect
from PIconnect.PIConsts import SummaryType

pi_tag = 'xxxxx'

with PI.PIServer() as server:
    points = server.search(pi_tag)[0]

points.summaries('*-1d', '*', '1h', SummaryType.TOTAL)
```

Our environments are pretty much the same except for minor version of AF SDK:
Python==3.9.13
PIconnect==0.10.4
Colleague's OSIsoft(r) AF SDK Version: 2.10.8.440 (not working)
My OSIsoft(r) AF SDK Version: 2.10.5.9050 (working)

Thanks

**Comment by Hugovdberg** [comment]
That is some weird behaviour, so here's a couple of things I want to check:
- Is it possible to post a more complete stack trace so I can see at which point the error occurs? 
- What version of the `pythonnet` package do you use?
- And are you sure that your colleague uses `PIconnect.PIConsts.SummaryType` and not `PIconnect.AF.Data.AFSummaryTypes`? It seems strange to me that it tries to convert a value from the AF SDK into an int.
- Just as a check, is the SDK correctly loaded for your colleague? Version 0.10.x should throw an exception if the SDK is not found, unless it detects it is imported in GitHub Actions or an Azure Pipeline, as I cannot publish the SDK itself. If it shows the correct SDK version upon import then you should be good, but I'm not 100% sure that I use the correct heuristics to detect the Azure pipeline.
 
According to the documentation, the `AFSummaryTypes` enumeration should be available: https://docs.osisoft.com/bundle/af-sdk/page/html/T_OSIsoft_AF_Data_AFSummaryTypes.htm, although it does not explicitly list version 2.10.8, but it shows 2.10.5 and 2.10.9 and no message that it had been gone for a release or a deprecation notice.

**Reply by Miigaa1191** [reply]
Thanks for your reply. Here is a screenshot: 
![image](https://user-images.githubusercontent.com/11828663/209262607-94752915-937c-4e75-996e-cbcfac6b04d4.png)
```pythonnet``` version is 3.0.1

**Comment by Hugovdberg** [comment]
Could you try this code:

```python
import importlib.metadata as impmeta

import PIconnect
import pythonnet

print(f"PIconnect version: {PIconnect.__version__}")
print(f"PI AF SDK version: {PIconnect.AF_SDK_VERSION}")
print(f"pythonnet version: {impmeta.version('pythonnet')}")

from System import DayOfWeek
print(int(DayOfWeek.Sunday))
```

It appears there was a bug in pythonnet that conversion from a .NET enum to integer does not work (see https://github.com/pythonnet/pythonnet/issues/1810). This should have been fixed some time ago, but this looks suspiciously a lot like that bug. Therefore the code above also tries to convert a simple .NET enum, which is also explicitly tested by the `pythonnet` developers to keep track of this bug. If that fails for your colleague too, then I think it is best to check whether `pythonnet` is really up to date with the latest version, and otherwise open an issue at the `pythonnet` repo.
If that code does work it looks like the AF SDK has some sort of undocumented change, but I find it kind of hard to believe that enums in .NET should suddenly no longer be represented by integers under the hood.

**Reply by Hugovdberg** [reply]
@Miigaa1191 have you tried the above already? I'm curious if there is still something we need to catch in the `PIconnect` code, or that it indeed was an issue with `pythonnet`.
